Formatting, grammar and spelling corrections.

mennis opened the pull request:
## Description

Formatting, grammar and spelling corrections.

## Checklist

<!--
  Please review the requirements for each checkbox, and check them
  off (change "[ ]" to "[x]") as you verify that they are complete.
-->
None of these are relevant as there are no functional changes.  Is there something I can to to provide confidence?

 - [ ] I made sure to update `./CHANGELOG.md`.
 - [ ] I made sure to add any docs changes required for my change (including release notes).
 - [ ] My change is adequately tested.
 - [ ] I updated `DEVELOPING.md` with any any special dev tricks I had to use to work on this code efficiently.
 - [ ] I updated `TELEMETRY.md` if I added, changed, or removed a metric name.
 - [X ] Once my PR is ready to have integration tests ran, I posted the PR in #telepresence-dev in the datawire-oss slack so that the "ok to test" label can be applied.

Commit:
Check in the telepresence-chart.tgz in the git repo

This file is embedded and needs to be included in the vendor directory
in downstream components. It must be included in the repository.

This commit changes `make build` so that it generates a temporary
`telepresence-chart.tgz` used for that build only (and hence embedded).
Once embedded, `git restore` is called to get rid of the changes. A new
`make build-version` target is added that doesn't restore the file.
The `build-version` target is used when pushing a new executable or
when doing `prepare-release`.

Signed-off-by: Thomas Hallgren <thomas@datawire.io>

Commit:
Don't run Test_APIServerIsProxied on host with covering default route

When the cluster runs in a VM on the same host as Telepresence and that
VM uses a bridged network, and that bridged network uses the same
default gateway, then it's not possible to "also-proxy" that gateway.

Signed-off-by: Thomas Hallgren <thomas@datawire.io>

thallgren reviewed (approved):
Well needed changes for sure. Thank you for submitting this.

I approve the PR but can you please add a "Signed off by" trailer at the end of the commit message?

Commit:
Move go source tar.gz to build-output and get rid of .gitignore for it

Signed-off-by: Thomas Hallgren <thomas@datawire.io>

Commit:
Use echo instead of printf to get consistent behavior

On a linux, the echo produces the "\n" at the end verbatim.

Signed-off-by: Thomas Hallgren <thomas@datawire.io>

Commit:
Use go package github.com/pkg/sftp instead of sftp-server binary

Special thanks to @ark3 for pointing in the right direction.

Signed-off-by: Thomas Hallgren <thomas@datawire.io>

Commit:
Avoid setting UID in agent at all times

Signed-off-by: Thomas Hallgren <thomas@datawire.io>

Commit:
Restructure build so that binary is built consistently on all platforms

This commit replaces the `ko` based build with a three-step docker
build. All steps are based on Alpine.

1. `tel2-base` target builds the traffic component, which is used both
   as a traffic-manager, agent init-container, and agent. The step uses
   `go mod download` to ensure that dependencies are cached by docker.
2. `tel2-build` Builds the `traffic` image from source.
3. `tel2` starts with clean alpine image and copies the necessary built
   artifacts from `tel2-build`

Everything related to `ko` is removed from the repository.

Signed-off-by: Thomas Hallgren <thomas@datawire.io>

mennis commented:
"Signed off by" updated.

thallgren commented on cmd/traffic/cmd/manager/internal/watchable/generic.tmpl.go:
```diff
@@ -4,31 +4,29 @@
+    "sync"
 
-	"google.golang.org/protobuf/proto"
-
-	"VALPKG"
```
This change is incorrect. Please remove it.

mennis replied:
Done

thallgren commented on cmd/traffic/cmd/manager/internal/watchable/generic_test.tmpl.go:
```diff
@@ -4,392 +4,391 @@
 
-	"github.com/stretchr/testify/assert"
+    "github.com/stretchr/testify/assert"
 
-	"VALPKG"
```
Also incorrect. Please remove this change.

mennis replied:
On it.  I am watching the tests too.

mennis commented:
Accidental code change found. Resetting to `sigNames := []string{}`

thallgren commented:
@mennis I think you are trying to edit the template files using an IDE and the IDE removes things it doesn't like on save. You'll need to use another editor when addressing the two comments that I made.

thallgren commented:
Also, since you have changed some files that affect how files are generated, you'll need to run `make generate` and commit the result in order to re-generate and get rid of the inconsistencies that the Licence verification complains about.

mennis committed:
Formatting, grammar and spelling corrections.

Plus required regenerated files.

Signed-off-by: Michaelian Ennis <mennis@corvus.net>

mennis commented:
Is the list of @thallgren's changes being visible in this conversation an indication that I've mis-merged something?

thallgren commented on pkg/client/userd/trafficmgr/traffic_manager.go:
```diff
@@ -751,7 +751,7 @@ func (tm *TrafficManager) getOutboundInfo(ctx context.Context) *daemon.OutboundI
 	// the cluster, since an open tunnel to the traffic-manager (via the API server) is itself required
 	// to communicate with the cluster.
-	neverProxy := []*manager.IPNet{}
+	var neverProxy []*manager.IPNet
```
This change is unnecessary and triggers a lint error.

mennis replied:
Fixed and lint is clean now.

thallgren reviewed (changes requested):
You might want to consider using `make lint` before pushing the commit to ensure that the code compiles and lints without errors.
